language/diem-framework/modules/DesignatedDealer.move

```diff
@@ -55,42 +48,21 @@ module DesignatedDealer {
 
     /// The `DesignatedDealer` resource is in an invalid state
     const EDEALER: u64 = 0;
-    /// The maximum number of tiers (4) has already been reached
-    const EINVALID_TIER_ADDITION: u64 = 1;
```

Comment:
Are we OK to reuse move error codes going forward? Or are we treating them as part of our compatibility set?

Reply:
Good question...I am of the opinion that we can reuse these as the Move explain tool will pickup these changes. Additionally, only the error categories are guaranteed to be stable, whereas the reasons (these numbers) don't have the same stability guarantees. Interested in @sblackshear's  thoughts on this as this is the first real case where we've deleted a lot of error reasons.

Reply:
Yes, very good question... I think reusing seems harmful and we should avoid it. In particular, I'm thinking of a case where a client has an SDK with the old error codes, but we have changed the meaning of (e.g.) error code 1 in the current version of the framework. For example, if 1 used to mean "insufficient funds" and it now means "recipient does not exist", the client with the old SDK may react to the error code in the wrong way.

In terms of what we should do in this PR, I guess what needs to be decided is a convention for "reserving" old error codes that were previously used and should not be reused in the future:
- Comment them out + note that they are deprecated. By convention, new error codes will always be added to the "end" (I think this was already the case)
- Leave them in, but change the names so they will no longer show up as error codes for `move explain` (e.g., `DEPRECATED_EINVALID_TIER_ADDITION`

I have a slight bias for the first one, particularly if we are going to expose constants in the transaction builders (this has been recently requested for `Roles`). We could have the transaction builder ignore constants named `DEPRECATED_*`, but that doesn't seem nice. Thoughts? Alternatives?

Reply:
I have the same opinion - also, these error codes will be dead code so they won't be used by clients in any case? commenting them is better imo as I would class changing codes as a breaking change